controllers/clustertopology_controller_compute_test.go

```diff
@@ -250,7 +252,281 @@ func TestGetClass(t *testing.T) {
+				controlPlaneInfrastructureMachineTemplate,
+			},
+			// Expecting valid return with valid control plane state, but no control plane infrastructure, cluster infrastructure or machine deployment state defined.
+			want: &clusterTopologyState{
+				cluster:               newFakeCluster(metav1.NamespaceDefault, "cluster1").withControlPlaneRef(objToRef(controlPlaneWithInfra)).Obj(),
+				controlPlane:          controlPlaneTopologyState{object: controlPlaneWithInfra, infrastructureMachineTemplate: nil},
+				infrastructureCluster: nil,
+				machineDeployments:    emptyMachineDeployments,
+			},
+		},
+		{
+			name:    "Cluster with cluster infrastructure reference control plane reference and control plane infrastructure",
+			cluster: newFakeCluster(metav1.NamespaceDefault, "cluster1").withInfraRef(objToRef(infraCluster)).withControlPlaneRef(objToRef(controlPlaneWithInfra)).Obj(),
+			class:   *clusterClassWithControlPlaneInfra,
+			objects: []client.Object{
+				infraCluster,
+				clusterClassWithControlPlaneInfra,
+				controlPlaneInfrastructureMachineTemplate,
+				controlPlaneWithInfra,
+			},
+			// Expecting valid return with valid control plane, control plane infrastructure and cluster infrastructure state, but no defined machine deployment state.
+			want: &clusterTopologyState{
+				cluster:               newFakeCluster(metav1.NamespaceDefault, "cluster1").withInfraRef(objToRef(infraCluster)).withControlPlaneRef(objToRef(controlPlaneWithInfra)).Obj(),
+				controlPlane:          controlPlaneTopologyState{object: controlPlaneWithInfra, infrastructureMachineTemplate: controlPlaneInfrastructureMachineTemplate},
+				infrastructureCluster: infraCluster,
+				machineDeployments:    emptyMachineDeployments,
+			},
+		},
+		{
+			name:    "Cluster with machine deployment state but no other states defined",
+			cluster: newFakeCluster(metav1.NamespaceDefault, "cluster1").Obj(),
+			class:   *clusterClassWithControlPlaneInfra,
+			objects: []client.Object{
+				infraCluster,
+				clusterClassWithControlPlaneInfra,
+				controlPlaneInfrastructureMachineTemplate,
+				controlPlaneWithInfra,
+				machineDeploymentInfrastructure,
+				machineDeploymentBootstrap,
+				machineDeploymentInCluster,
+			},
+			// Expecting valid return with valid control plane, control plane infrastructure and cluster infrastructure state, but no defined machine deployment state.
+			want: &clusterTopologyState{
+				cluster:               newFakeCluster(metav1.NamespaceDefault, "cluster1").Obj(),
+				controlPlane:          controlPlaneTopologyState{},
+				infrastructureCluster: nil,
+				machineDeployments:    []machineDeploymentTopologyState{{object: machineDeploymentInCluster, bootstrapTemplate: machineDeploymentBootstrap, infrastructureMachineTemplate: machineDeploymentInfrastructure}},
+			},
+		},
+		{
+			name:    "Class assigning control plane infrastructure and cluster with control plane reference but no control plane infrastructure",
```

Comment:
I'm not sure I fully understand this use case

Reply:
This is the case where a clusterClass describes a control plane with an infrastructure reference, but the control plane found in the API server has no infrastructure reference. This is assumed to be some error in configuration by the user so it errors out here.

Reply:
Thanks!